src/function/once.ts

```diff
@@ -15,20 +15,16 @@
  * initialize(); // Logs: 'Initialized!' and returns true
  * initialize(); // Returns true without logging
  */
-export function once<F extends () => any>(func: F): F {
+export function once<F extends (...args: unknown[]) => any>(func: F): () => ReturnType<F> {
```

Comment:
```suggestion
export function once<F extends () => any>(func: F): () => ReturnType<F> {
```
I think `func` should be a unary function.

Reply:
If `func` is unary, why can't `once` just return `F` here?

Reply:
> I think `func` should be a function accepting no arguments.
``` javascript
function say(str) {
  console.log(str)
}

const onced = once(say)
onced("hello")
```

In this case, the say function requires an argument, so func should not be restricted to functions that accept no arguments.

Reply:
Also, even if the interface is updated, it shouldn’t cause any issues with backward compatibility.

Reply:
> If `func` is a function accepting no arguments, why can't `once` just return `F` here?

I’ve applied this to the code. Thank you!

Reply:
If we allowed functions accepting arguments, we might end up with this code:

```tsx
function sum(x, y) {
  return x + y;
}

const onced = once(sum);
sum(1, 2) // 3
sum(2, 4) // 3
```

which is very awkward. So I thought we should restrict our function to functions which accept no arguments...

Reply:
> If we allowed functions accepting arguments, we might end up with this code:
> 
> ```tsx
> function sum(x, y) {
>   return x + y;
> }
> 
> const onced = once(sum);
> sum(1, 2) // 3
> sum(2, 4) // 3
> ```
> 
> which is very awkward. So I thought we should restrict our function to functions which accept no arguments...

This function returns a new function, so a situation like the one above is unlikely to occur..?

Reply:
When searching through the use cases of `once`, I think there are two usecases:

**[1] Ensuring that we initialize certain objects only once, like:**

```tsx
const createClient = once(() => new Client());

createClient();
createClient(); // client is created only once
```

**[2] Ensuring that we make a side effect only once, like:**

```tsx
const log = once(console.log);

log('hello')
log('world') // this call is ignored
```

But I guess [1] is covered with [memoize](https://es-toolkit.slash.page/reference/function/memoize.html#memoize), so `once` might focus on solving the problem [2].

Reply:
If we were to support [2], we might restrict the function to return `void` when it receives parameters, like:

```tsx
export function once<F extends () => any>(func: F): F;
export function once<F extends (...args: any) => void>(func: F): F;
```

Reply:
> If we were to support [2], we might restrict the function to return `void` when it receives parameters, like:
> 
> ```tsx
> export function once<F extends () => any>(func: F): F;
> export function once<F extends (...args: any) => void>(func: F): F;
> ```

Oh, if we change it like this, the function’s role will become much clearer!

Reply:
But since we need backward compatibility, the best option we can choose might be supporting both:

```tsx
export function once<F extends () => any>(func: F): F;
export function once<F extends (...args: any) => void>(func: F): F;
```

Reply:
> But since we need backward compatibility, the best option we can choose might be supporting both:
> 
> ```tsx
> export function once<F extends () => any>(func: F): F;
> export function once<F extends (...args: any) => void>(func: F): F;
> ```

I added these suggestions to the code.